### PR TITLE
New Post: Automating Designer-to-Developer Handoff

### PR DESCRIPTION
**Pull request by r10-gchiong**
## Type
- [X] **Editorial:** This adds or updates an article, or assets associated with an article.

## Description

This post is mostly about a CLI tool we built internally for the EC-Mobile app projects for handling the design deliverable from Figma. I was initially asked to write an article about it on [R-hack](https://commerce-engineer.rakuten.careers/entry/en/tech/0045). But I have since been requested to submit it here as well for your consideration.

## Editorial Checklist
- [X] This introduces a new article.
  - [X] My article is original content: I did not copy or plagiarize another source.
  - [X] I am fine with publishing this content under the [Creative Commons — Attribution 4.0 International](https://creativecommons.org/licenses/by/4.0/) license.
  - [X] If my content includes assets I didn't create myself, e.g. visuals, I confirmed that I have the right to use them.
  - [X] I have checked my article for spelling and grammatical mistakes.
  - [X] I have checked my article for technical mistakes and I believe it's technically sound.
  - [X] I have submitted my article for a [Dev PR Review](https://confluence.rakuten-it.com/confluence/x/NJXtlQ). The ticket is at [DAF-2476](https://jira.rakuten-it.com/jira/browse/DAF-2476).

### CODE REVIEW

**Comment by jcayzac** [comment]
> LGTM but I'm waiting for #31 to be merged before I approve this PR.

Done. @r10-gchiong could you please rebase?

**Comment by r10-gchiong** [comment]
> > LGTM but I'm waiting for #31 to be merged before I approve this PR.
> 
> Done. @r10-gchiong could you please rebase?

@jcayzac I just rebased.

File: content/post/rds-automating-designer-to-developer-handoff/index.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+Hi, this is Gian from the E-Commerce Mobile Architect Group. The mission of our group is to improve productivity by creating tools to automate tasks and providing technical support for the e-commerce mobile app development teams. In this article, we will discuss about one of the developer tools we created to automate designer-to-developer handoff.
+
+# Overview
+
+Designer-to-developer handoff is an important part of the product development life cycle. It takes place when the design deliverables are ready to be provided to developers for implementation.
+
+In the E-Commerce Mobile App Development Department, the design deliverables are mainly created using [Figma](https://www.figma.com). The design in Figma is considered as the single-source-of-truth from which the app development team derives the product's UI. As such, it is critical to ensure that it is formatted in a way that can be clearly understood by the developers to avoid any inconsistencies between the design and the final product.
+
+# Design Deliverable Prerequisites
```
Comment (jcayzac) [inline content/post/rds-automating-designer-to-developer-handoff/index.md:22]: All the headers are level 1 (`#`), which is normally reserved for the document's title. Could you change them to level 2 (`##`) ?

Comment (r10-gchiong) [reply content/post/rds-automating-designer-to-developer-handoff/index.md]: No prob. Updated.

File: content/post/rds-automating-designer-to-developer-handoff/index.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+authors:
+  - Gianne Chiong
+hero: /post/rds-automating-designer-to-developer-handoff/images/rds_logo.png
+draft: true
+license: cc-by
+seo:
+  canonical: https://commerce-engineer.rakuten.careers/entry/en/tech/0045
+---
+
+Hi, this is Gian from the E-Commerce Mobile Architect Group. The mission of our group is to improve productivity by creating tools to automate tasks and providing technical support for the e-commerce mobile app development teams. In this article, we will discuss about one of the developer tools we created to automate designer-to-developer handoff.
```
Comment (NemoOudeis) [inline content/post/rds-automating-designer-to-developer-handoff/index.md:14]:
```suggestion
Hi, this is Gian from the E-Commerce Mobile Architect Group. The mission of our group is to improve productivity by creating tools to automate tasks and providing technical support for the e-commerce mobile app development teams. In this article, we will discuss one of the developer tools we created to automate designer-to-developer handoff.
```

File: content/post/rds-automating-designer-to-developer-handoff/index.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+Designer-to-developer handoff is an important part of the product development life cycle. It takes place when the design deliverables are ready to be provided to developers for implementation.
+
+In the E-Commerce Mobile App Development Department, the design deliverables are mainly created using [Figma](https://www.figma.com). The design in Figma is considered as the single-source-of-truth from which the app development team derives the product's UI. As such, it is critical to ensure that it is formatted in a way that can be clearly understood by the developers to avoid any inconsistencies between the design and the final product.
+
+## Design Deliverable Prerequisites
+
+Our designers prepare the design deliverables following a few general guidelines. Here are a couple of examples.
+
+- Assets that are intended to be exported into an image format are converted into a Figma Component and laid out in a separate page and/or frame. Likewise, typography and color styles are also defined as Figma Styles, and representations of these styles are also shown in dedicated pages and/or frames.
```
Comment (NemoOudeis) [inline content/post/rds-automating-designer-to-developer-handoff/index.md:26]:
```suggestion
- Assets that are intended to be exported into an image format are converted into a Figma Component and laid out in a separate page and/or frame. Likewise, typography and color styles are defined as Figma Styles, and representations of these styles are also shown in dedicated pages and/or frames.
```

File: content/post/rds-automating-designer-to-developer-handoff/index.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+## Further Expansion
+
+By using an intermediary format (RDL), we can add support for other design files by only modifying the steps leading up to the conversion of the design file into the RDL format. For instance, if we were to add support for Sketch files, theoretically, only specific parts of the flow will have to change as shown in the diagram below.
+[![rds_flow_sketch](./images/rds_flow_sketch.png)](./images/automating-code-and-asset-generation-from/rds_flow_sketch.png)
+
+## Conclusion
+
+**RDS** is just one of the internal tools that we have developed to help us automate tasks that are otherwise done manually. It helps developers stay in sync with the design files created by our designers.
+It initially started as a one-off tool meant for our iOS projects under Rakuten Commerce TECH. However, it's now part of the workflow for our Android projects as well.
```
Comment (NemoOudeis) [inline content/post/rds-automating-designer-to-developer-handoff/index.md:143]:
```suggestion
It started as a one-off tool meant for our iOS projects under Rakuten Commerce TECH. However, it's now part of the workflow for our Android projects as well.
```

**Review by NemoOudeis — APPROVED** [review]

**Pull request merged**